core/utils/util.c

```diff
@@ -112,23 +112,29 @@ void _lf_message_print(const char* prefix, const char* format, va_list args,
     // interleaved between threads.
     // vprintf() is a version that takes an arg list rather than multiple args.
     char* message;
+    if (_lf_my_fed_id < 0) {
```

Comment:
This part gives compile time errors, since the uint16_t type will never be a negative integer.

However, if I remove the whole if and else state, `general_utils_pqueue_test_c` fails with a segfault.

It will not go through the removed if statement, and fail at line 129, because the `name` is a NULL pointer. Does anyone have ideas?

I'll leave the call stack below.
![image](https://github.com/lf-lang/reactor-c/assets/74869052/77c47745-c4d0-43ad-b67d-31b9be02d83f)

Reply:
I think it would be good to replace `-1` with another id that indicates an uninitialized id or no id. Either `0x0000` or 0xffff.